flagsmith/flagsmith.py

```diff
@@ -332,7 +332,20 @@ def _get_environment_flags_from_document(self) -> Flags:
         if self._evaluation_context is None:
             raise TypeError("No environment present")
 
-        evaluation_result = engine.get_evaluation_result(self._evaluation_context)
+        # Omit segments from evaluation context for environment flags
+        # as they are only relevant for identity-specific evaluations
+        context_without_segments = typing.cast(
+            SDKEvaluationContext,
+            {
+                key: value
+                for key, value in self._evaluation_context.items()
+                if key != "segments"
+            },
+        )
```

Comment:
This helps to avoid both the cast and a bytecode loop:

```suggestion
        context_without_segments = self._evaluation_context.copy()
        context_without_segments.pop("segments", None)
```